Fork branch cetko refactor

sasakalaba opened the pull request.

angstwad committed:
Add Verify Email message flashing

gcetusic committed:
added stormpath to requirements.txt

gcetusic committed:
dynamic form settings based on environment and settings files - work in progress

gcetusic committed:
tests for custom settings object

gcetusic committed:
replaced several old style settings in templates with new ones -> commented the rest

Conflicts:
	flask_stormpath/__init__.py

Commented out StormpathSettigns.__contains__, current mapping breaks the code.

gcetusic committed:
misc fixes to make standard forms show

gcetusic committed:
fixed registration

Conflicts:
	flask_stormpath/views.py

Removed field mapping and replaced with Resource.to_camel_case().

gcetusic committed:
change logout redirect based on settings

gcetusic committed:
misc work on verification flow

Conflicts:
	flask_stormpath/__init__.py

Updated test_settings.TestCheckSettings:

 - updated imports
 - updated check_settings method call
 - updated skipped test error messages

igorrcosta committed:
Fix the base screen style.

This modification will center the login view container. See http://igorrcosta.pythonanywhere.com/login on any screen size for an example of the bug.
This change was already made on the express-stormpath library:
https://github.com/stormpath/express-stormpath/commit/77226f7bdf1fb728d2b1c703fabe042436e4747c

rdegges committed:
Merge pull request #62 from igorrcosta/patch-1

Fix the base screen style.

gcetusic committed:
first stab at me view

gcetusic committed:
don't lazily load stormpath client and application since it's already loaded during strategy enriching

gcetusic committed:
moving imports to a more appropriate location

gcetusic committed:
moved json representation login to stormpath python sdk

rdegges committed:
Upgrading facebook-sdk dependency to 1.0.0.

This provides full Python 3 support, as well as fixes compatibility issues with
PythonAnywhere and other web hosts who require outbound HTTP proxying.

rdegges committed:
Version bump.

rdegges committed:
Adding docs for upgrading / changes.

gcetusic committed:
json vs html response basic functionality

gcetusic committed:
error handling for login and register

gcetusic committed:
registration via json

gcetusic committed:
autologin if verify

gcetusic committed:
basePath setting taken into account

Commit:
Skipping broken tests.

 - importing stormpath_config from a local repository (not available through
   pip)
 - commented out MoveApiKeyToClientAPIKeyStrategy (not yet implemented in
   stormpath_config)
 - fixed field mapping in app.config dict object
 - skipped all the tests that are broken due to partly implemented
   StormpathSettings class

Commit:
Updated commit:

	'moved init settings to manager because we need the manager settings'
	(c73eb55fc45b155aade965294f7724e61c052c16)

Updated test_settings.py.

Commit:
Merge branch 'develop' into temp_fork_cetko

Conflicts:
	requirements.txt
	setup.py

Commit:
Updated commit:
	'removed redundant settings -> defined in default config'
	(e7e1ad23f6ceaf829747621d374a4e7347e7a7a1)

Commented out STORMPATH_BASE_TEMPLATE config setting.

Commit:
Updated broken test messages.

ohadperry committed:
raising a clear error on failing to find application

ohadperry committed:
better error

Commit:
Updated error messages on skipped tests.

Commit:
Updated check_settings.

 - check_settings now validates data through the new StormpathSettings class.
 - assertRaises now called as 'with' context manager
 - added ConfigurationError message check

Twista committed:
replace deprecated import flask.ext.* with new flask_*

rdegges committed:
Merge pull request #68 from Twista/master

replace deprecated import flask.ext.* with new flask_*

Commit:
Added temporary fixes for form instantiation.

 - double form instantiation
   (this needs to be fixed A.S.A.P. !!)
 - added the AppWrapper for manipulating Accept
   headers when testing views

Commit:
Updated registration view.

 - extended 'enabled/required' check with surname and middle name
 - optional_field properties are now collected through a for loop
 - replaced 'UNKNOWN' with 'Anonymous'
 - replaced form.data with data (since you cannot assign new values to
   form.data)
 - added tearDown to test_views (for deconstructing dynamically set
   StormpathForm attributes)
 - fixed data indentation
 - fixed redirect_url

sasakalaba commented on tests/helpers.py:
```diff
@@ -14,6 +14,10 @@
+# FIXME: setup a better way to load environment variables
+environ['STORMPATH_API_KEY_ID'] = '15O7VLV850461TYBRFP91KRR4'
+environ['STORMPATH_API_KEY_SECRET'] = '8Ao/UesWQVhVkE7LL7ZVApHWn/r0cygrrHaruh75ipk'
+
```
Temporary fix dok ne rijesim bitnije stvari. Ovo cu maknut cim prije.

sasakalaba commented on flask_stormpath/__init__.py:
```diff
@@ -250,6 +250,21 @@ def check_settings(self, config):
+        ]):
+            raise ConfigurationError(
+                'You must define your Stormpath application.')
 
```
Ovo moram maknuti jer smo rekli na reviewu da se ovo vadi iz ValidateClientConfigStrategy.

sasakalaba commented on flask_stormpath/views.py:
```diff
@@ -177,6 +179,8 @@ def login():
+    # FIXME ASAP: cannot initialize StormpathForm properly
+    form = LoginForm()
     form = LoginForm(csrf_enabled=False)
 
```
Ovog me sram, but alas nisam vise znao kaj da radim sa tom formom, pa sam stavio ovo da mogu dalje view-ove raditi. Probleme sa formama cemo proci u detalje u srijedu.

sasakalaba commented on tests/test_views.py:
```diff
@@ -7,15 +7,30 @@
+            'application/xml;')
+        return self.app(environ, start_response)
+
+
```
Ovako sam ubacio Accept header u request buduci da ga `request_wants_json()` provjerava u view-ovima.

sasakalaba commented on flask_stormpath/views.py:
```diff
@@ -72,26 +72,35 @@ def register():
+                    data[field] = 'Anonymous'
+            else:
+                if not data[field] and not field_properties[field]['required']:
+                    data[field] = 'Anonymous'
```
Provjeravam opcionalne form fieldove i u slucaju da su enabled i nemaju vrijednost, postavljam na 'Anonymous', buduci da se proslijediti na `User.create(**data)` te moraju imati vrijednost razlicitu od None ili blank.

avojnovicDk commented on flask_stormpath/__init__.py:
```diff
@@ -122,6 +135,181 @@ def init_app(self, app):
+            client_factory=lambda client: self.client)
+        eifrcs.process(self.app.config['stormpath'].store)
+
+        self.application = self.client.applications.get(
```
This Stormpath application.

avojnovicDk commented on flask_stormpath/__init__.py:
```diff
@@ -97,11 +109,11 @@ def init_app(self, app):
         # configured.
-        check_settings(app.config)
+        self.check_settings(app.config)
 
```
`self.init_stormpath(app.config)` koji inicijalizira Stormpathov `client` i `application` (iz `init_settings` funkcije)

avojnovicDk commented on flask_stormpath/__init__.py:
```diff
@@ -97,11 +109,11 @@ def init_app(self, app):
         # Check our user defined settings to ensure Flask-Stormpath is properly
         # configured.
-        check_settings(app.config)
+        self.check_settings(app.config)
```
We don't need this because we validated settings in `init_settings`.

avojnovicDk commented on flask_stormpath/views.py:
```diff
@@ -138,10 +149,12 @@ def register():
                     'stormpath']['web']['register']['nextUri']
                 if not redirect_url:
-                    redirect_url = current_app.config[
+                    login_redirect = current_app.config[
```
web_config.get('register', web_config.get('login', '/'))

Commit:
Updated login view.

 - fixed flash error message output
 - added setUp to TestLogin

Commit:
Updated gitignore.

Commit:
Updated .gitignore #2.

 - ignoring html coverage report

Commit:
Combined redirect tests in test_views.

Combined test_redirect_to_login_and_register_url and
test_redirect_to_register_url into one test, since they test
the same logic.

Commit:
Fixed minor issues in register view.

 - simplified redirect priority logic
 - optimized 'Anonymous' setting on disabled form fields

Commit:
Fixed dynamic form building.

Forms are now build dynamically through the StormPath form.

 - StormpathForm now builds the form fields via classmethod.
 - removed RegisterForm and LoginForm (replaced with StormpathForm)

Commit:
Removed fail flag from registration view.

- 'fail' flag checks for blank fields who are both enabled and required.
  Since form.validate_on_submit() already does the exact same thing, we've
  removed the `fail` flag

Commit:
Added if request == POST check to registration view.

Commit:
Applied confirm_password check to registration view.

 - moved confirm password error message from StormpathForm
   to register view

Commit:
Moved validation logic from register view to StormpathForm.

Commit:
Fixed autologin on register.

Commit:
Config settings for login and register are now accessed more easily.

avojnovicDk commented on flask_stormpath/views.py:
```diff
@@ -65,68 +64,51 @@ def register():
+            # We'll just set the field values to 'Anonymous' if the user
+            # has explicitly said they don't want to collect those fields.
+            for field in ['given_name', 'surname']:
+                if field not in data or not data[field]:
```
`if not data.get(field)`:

Commit:
Fixed dynamic field building on StormpathForm.

 - added a subclass in StormpathForm (so as to keep the original
   class unaltered)
 - renamed method append_fields() to specialize_form()
 - added an Email validator
 - added tests for StormpathForm
 - moved some of the error messages from test_views to test_forms
 - added confirmPassword test to test_views
 - removed tearDown method from test_views TestRegister
 - removed redundant imports

Commit:
Fixed PEP8 errors.

Commit:
Fixed if statement.

Commit:
Updated TestLogout tests.

 - requests are now html/text type.

Commit:
Updated forgot view.

 - added tests for forgot view
 - added request_wants_json / make_stormpath_response to forgot view
 - make_stormpath_response now renders passed template, not login

Commit:
Minor test_views refactoring.

 - moved user creation and AppWrapper to the new
   StormpathViewTestCase class (now a parent class to all
   test_views classes)

Commit:
Minot test_models refactoring.

 - moved user creation to setUp

Commit:
Added test for request_wants_json().

Commit:
Implemented json to forms.

 - added json property

Commit:
Added test_make_stormpath_response test.

Commit:
Added to_json method to User model.

Commit:
Moved 'account' key from views to to_json() method.

Commit:
Added json test to test_views.

 - StormpathViewTestCase can now dynamically set accept headers
 - TestHelperFunctions now inherit from StormpathViewTestCase
 - added assertJsonResponse() method to StormpathViewTestCase
 - moved check_header to StormpathViewTestCase
 - added json_response tests to every view test case
 - make_stormpath_response now accepts status codes
 - updated User test for json (added 'account' key)

viksit committed:
Error object in StormpathError is now a unicode string

From stormpath version 2.2.0 and later, the `flash` function should be simply passing `err.message` in, rather than think it's a dictionary object.

This is the stack trace without the fix,


```
Traceback (most recent call last):
  File "/usr/conda_env/lib/python2.7/site-packages/flask/app.py", line 2000, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/conda_env/lib/python2.7/site-packages/flask/app.py", line 1991, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/conda_env/lib/python2.7/site-packages/flask/app.py", line 1567, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/conda_env/lib/python2.7/site-packages/flask/app.py", line 1988, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/conda_env/lib/python2.7/site-packages/flask/app.py", line 1641, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/conda_env/lib/python2.7/site-packages/flask/app.py", line 1544, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/conda_env/lib/python2.7/site-packages/flask/app.py", line 1639, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/conda_env/lib/python2.7/site-packages/flask/app.py", line 1625, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/conda_env/lib/python2.7/site-packages/flask_stormpath/views.py", line 98, in register
    flash(err.message.get('message'))
AttributeError: 'unicode' object has no attribute 'get'
```

## Versions
```
Flask-Stormpath==0.4.5
stormpath==2.4.0
Flask=0.10.1
```

Commit:
Added me tests.

Commit:
Templates now center aligned.

Commit:
Removed hardcoded api id and secret from tests.

Commit:
Removed the ugly sys import.

Commit:
Fixed remaining settings tests.

 - fixed social login validation
 - fixed the dummy test(test_register_default_account_store)
 - removed setpUp and tearDown(since we already need apiKey.properties file
   to load our environ variables)

Commit:
Removed backslashes for newlines.

Commit:
Fixed the remainder of broken tests.

 - removed every skip decorator
 - moved AppWrapper from test_views to helpers
 - removed obsolete imports

Commit:
Renamed AppWrapper to HttpAcceptWrapper.

Commit:
Minor test refactoring.

 - added explicit hardcoded values to a couple of tests

Commit:
Fixed form.errors output on register and login views.

 - json response on GET register now returns form fields (not errors)
 - json response now returns proper form errors on

Commit:
Updated forgot view.

 - form field settings now in config file
 - removed ForgotPasswordForm (now using StormpathForm)
 - updated json support (updated messages and added tests)

Commit:
Fixed status code in views.

Commit:
Added function description to views.

Commit:
Updated change view.

 - form field settings now in config file
 - removed ChangedPasswordForm (now using StormpathForm)
 - updated json_support (and tests)
 - updated forgot_change template

Commit:
Updated form tests.

 - moved form building tests to a custom assert method
 - added form building tests for forgot and change forms

Commit:
Updated fields on User.to_json.

Commit:
Minor update on tests for User.to_json.

Commit:
Updated config file.

 - added templateSucces values to forgot and change views settings
   (secondary template setting)

avojnovicDk commented on flask_stormpath/__init__.py:
```diff
@@ -267,23 +264,31 @@ def check_settings(self, config):
+            if 'STORMPATH_SOCIAL' in config:
+                google_config = config['STORMPATH_SOCIAL'].get('GOOGLE')
+            else:
+                google_config = None
```
`google_config = config.get('STORMPATH_SOCIAL', {}).get('GOOGLE', {})`

avojnovicDk commented on flask_stormpath/__init__.py:
```diff
@@ -267,23 +264,31 @@ def check_settings(self, config):
             if not google_config or not all([
                 google_config.get('client_id'),
                 google_config.get('client_secret'),
             ]):
```
No need for `not google_config` if `google_config` is a dict.

avojnovicDk commented on flask_stormpath/models.py:
```diff
@@ -73,14 +74,24 @@ def delete(self):
+        for key in writable_attrs:
+            attr = getattr(self, key, None)
+            json_data['account'][key] = (
+                attr if not isinstance(attr, datetime) else attr.isoformat())
```
`json_data = {'account': {attr: getattr(self, attr, None) for attr in attrs}}`

avojnovicDk commented on flask_stormpath/models.py:
```diff
@@ -73,14 +74,24 @@ def delete(self):
+            attr = getattr(self, key, None)
+            json_data['account'][key] = (
+                attr if not isinstance(attr, datetime) else attr.isoformat())
+        return json.dumps(json_data)
```
`return json.dumps(json_data, default=datetime_handler)`

avojnovicDk replied:
http://stackoverflow.com/questions/23285558/datetime-date2014-4-25-is-not-json-serializable-in-django

avojnovicDk commented on flask_stormpath/models.py:
```diff
@@ -73,14 +74,24 @@ def delete(self):
-        return json.dumps({'account': {
-            key: getattr(self, key, None) for key in writable_attrs}})
+            'full_name'
+        )
```
needs better naming - these are not (only) writable attributes

avojnovicDk commented on flask_stormpath/__init__.py:
```diff
@@ -198,8 +195,8 @@ def init_settings(self, config):
-                    key in config['stormpath']:
+            if (key.startswith(config['stormpath'].STORMPATH_PREFIX) and
+                    key in config['stormpath']):
                 config['stormpath'][key] = value
```
This does nothing.

avojnovicDk commented on flask_stormpath/__init__.py:
```diff
@@ -267,23 +264,31 @@ def check_settings(self, config):
+            if 'STORMPATH_SOCIAL' in config:
+                facebook_config = config['STORMPATH_SOCIAL'].get('FACEBOOK')
+            else:
+                facebook_config = None
```
same as with Google

avojnovicDk commented on flask_stormpath/__init__.py:
```diff
@@ -267,23 +264,31 @@ def check_settings(self, config):
             ]):
-                raise ConfigurationError('You must define your Facebook app settings.')
+                raise ConfigurationError(
+                    'You must define your Facebook app settings.')
```
same as with Google

avojnovicDk commented on flask_stormpath/views.py:
```diff
@@ -32,22 +28,28 @@
     FACEBOOK = True
```
Just a reminder - this has changed in the meantime.

Commit:
Refactored views.

 - views are now class based
 - updated test_error_messages for each view
   (some test cases were missing, and some are already tested in test_forms)
 - json responses on StormpathError now all return status (instead of error)
 - added a json response to logout view
 - renamed TestHelperFunctions to TestHelperMethods

Commit:
User.to_json now also returns expanded data.

 - renamed writtabe_attrs to attrs.
 - added an expand check

Commit:
Forms can now accept empty configs.

 - StormpathForm can now accept empty config files
   (and in turn will return emtpy forms).

Commit:
Added 406 check.

 - every view now checks the accept header in the validate_request method
   (if the request is not json or html, we'll return a 406)
 - refactored request_wants_json method
 - added additional tests for checking the accept header (and refactored
   existing ones)
 - LogoutView now calls super on init (since we need accept_header and
   allowed_types from the parent class)
 - moved header accept manipulation logic from test_views to helpers
   (StormpathTestCase), since we now need the headers present
   in context_processors and decorators tests

Commit:
Removed the obsolete VerificationForm.

Commit:
** WORK IN PROGRESS ** Social views refactor 1/2.

avojnovicDk commented on flask_stormpath/views.py:
```diff
@@ -555,27 +216,331 @@ def google_login():
+        self.config = config
+        self.form = (
+            StormpathForm.specialize_form(config['form'])()
+            if config else None)
```
Check if there is need for this (any case when `self.form` needs to be `None`?)

avojnovicDk commented on flask_stormpath/views.py:
```diff
@@ -555,27 +216,331 @@ def google_login():
+    def process_request(self):
+        """ Custom logic specialized for each view. Must be implemented in
+            the subclass. """
+        raise StormpathForm('You must implement process_request on your view.')
```
`raise NotImplementedError('Subclasses must implement this method.')`

avojnovicDk commented on flask_stormpath/views.py:
```diff
@@ -555,27 +216,331 @@ def google_login():
+        if self.request_wants_json():
+            return self.make_stormpath_response(data=current_user.to_json())
+
+        return redirect(request.args.get('next') or self.config['nextUri'])
```
use `if not` as in `RegisterView`.

rdegges committed:
Merge pull request #69 from viksit/patch-1

Change flash() in StormpathError to take in unicode string rather than dict

Commit:
User.to_facebook refactor.

 - moved the social directory creation logic from views to models
 - added redirect to facebook login in case of facebook login breaking
 - added tests for User.from_facebook method
 - added get_facebook_access_token to helpers (used to retrieve user access
   token needed for testing)

Commit:
Updated the way we handle test environment variables.

 - removed secret and id loading from apikey.properties file
   (now loaded from environment, set in pytest.ini)
 - added pytest-env package
 - added stormpath secret and id validation in helpers.py
 - added social credentials validation in EnvironmentSettings
   (for calls in test_models and test_views)
 - removed get_facebook_access_token (instead using environ.get())
 - renamed EnvironmentSettings to CredentialsValidator
 - updated all ids, secrets and access tokens (since they were visible in
   previous commits)

Commit:
User.to_google refactor.

 - moved the social directory creation logic from views to models
 - added tests for User.from_google method

Commit:
User social methods tests refactor 1/2.

 - removed access token from facebook tests (now using mocking instead)
 - removed access token validation
 - added validation for google client id and secret
 - moved social credentials check to helpers.py
 - added tear down of stormpath resources in credentials validation
 - moved StormpathTestCase.tearDown logic to destroy_resources function
 - updated error message testing on social methods

Commit:
User social methods tests refactor 2/2.

 - social flow in models is now tested in a separate class (a mixin)

Commit:
Moved user creation in our test to main tester class (StormpathTestCase).

Commit:
Updated models tests.

 - updated the correct message for test_from_social_invalid_access_token_
   with_existing_directory test case.
 - added save test
 - updated User.save method (removed return value, save now returns instance)

Commit:
Refactored from_social methods in models.

 - from_facebook and from_login now inherit their logic from from_social
   method
 - updated tests for models

Commit:
Removed function based google_login view.

Commit:
Added a testing structure for social views.

Commit:
Social views refactor 2/2.

 - social views now inherit from SocialViews
 - added tests for social views
 - moved social configuration in testing to bootstrap_flask_app

Commit:
Updated StormpathView.

 - self.form no longer None on init
 - replaced StormpathError with NotImplementedError on process_request method
 - updated redirect in login view (same as in register now)
 - SocialView and  MeView now inherit from flask View class
 - request_wants_json is now a property method
 - moved validate_request to init
 - logout doesn't check for json, but always redirects to login (who then
   checks for json)

Commit:
Added redirect logic to assertJsonResponse in test_views.

 - assertJsonResponse can now check for redirects and their data
 - updated json response test on get method for the LoginView

Commit:
Updated SocialView error message.

 - error message is now intended for users, not developers

Commit:
Updated the accept header 406 check.

 - the 406 check now returns the first allowed response type if the
   accept header is missing or */*

Commit:
Removed Python 3 specialized import.

 - facebook-sdk now supports Python 3

Commit:
Added Python 3 import for mocking.

Commit:
Added a custom datetime serializer to our User.to_json method.

Commit:
Pulled the latest version of the config file.

Commit:
Fixed PEP 8 errors in init.py.

Commit:
Updated imports.

Commit:
Updated templates and views based on the new config 1/3.

 - reordered view in the yaml file (consistent placing instead of random)
 - templates are now read from class views, not config file
 - renamed ChangeView to ChangePasswordView
 - renamed ForgotView to ForgotPasswordView
 - renamed templates:
      - forgot.html >> forgot_password.html
      - forgot_email_sent.html >> forgot_password_success.html
      - forgot_change.html >> change_password.html
      - forgot_complete.html >> change_password_success.html

Commit:
Updated templates and views based on the new config 2/3.

 - added a visible option to StormpathForm

rdegges committed:
Merge pull request #67 from ohadpartuck/better_errors

raising a clear error on failing to find application

rdegges committed:
Merge pull request #53 from angstwad/verify_message

Add "Verify Email" message flashing

rdegges committed:
Updating the changelog with release notes for 0.4.6.

rdegges committed:
Merge branch 'master' into develop

* master:
  Updating the changelog with release notes for 0.4.6.
  Error object in StormpathError is now a unicode string
  replace deprecated import flask.ext.* with new flask_*
  better error
  raising a clear error on failing to find application
  Adding docs for upgrading / changes.
  Version bump.
  Upgrading facebook-sdk dependency to 1.0.0.
  Fix the base screen style.

rdegges committed:
Version bump.

rdegges committed:
Whoops! Fixing merge markers.

rdegges committed:
Moving requirements into setup.py. Let setuptools deal with the madness =D

rdegges committed:
Updating travis install step.

Using pip now.

rdegges committed:
Simplifying codez.

rdegges committed:
Upgrading facebook-sdk to latest release (to get py3k support).

rdegges committed:
Ignoring pip cache dir.

rdegges committed:
Cleanup

rdegges committed:
Minor test cleanup.

rdegges committed:
Minor test cleanup.

rdegges committed:
Removing Facebook and Python3 errors.

We're using the new release now which supports python3 fully, so this old code
is no longer necessary.

rdegges committed:
Replacing old import syntax with new.

rdegges committed:
Minor cleanup.

rdegges committed:
Minor cleanup.

rdegges committed:
Upgrading Stormpath dependency to fix broken tests.

This release contains new error handling fixes.

rdegges committed:
Fixing broken test.

Commit:
Updated error processing in views.

 - also skipped check_settings test(soon to be obsolete)

01walid committed:
Convert `is_authenticated`, `is_active`, and `is_anonymous` to properties to conform to Flask-Login >=0.3.0

Commit:
Updated templates and views based on the new config 3/3.

 - removed 406 response for invalid request
 - request is now passed to the developer/framework, or returns 501 by default

rdegges committed:
Merge pull request #75 from 01walid/master

Convert `is_authenticated`, `is_active`, and `is_anonymous` to properties to conform with Flask-Login >=0.3.0

rdegges committed:
Adding changelog / upgrading docs.

rdegges committed:
Version bump.

jcass77 committed:
Use standard WTForm validation for 'register' form.

jcass77 committed:
Specify test folder to avoid all tests on PATH from being run.

jcass77 committed:
Add error messages for required email and password fields.

jcass77 committed:
Use standard email validators.

Add missing validation error messages.

jcass77 committed:
Fix field name reference.

rdegges committed:
Merge pull request #78 from BarrusInvestments/enhance/form_validation

Use standard WTForm validation for 'register' page.

Commit:
Implemented verify view 1/2 (WIP).

 - added a verify view
 - added a new template for resending verification tokens
 - updated config file to specify form fields for our new template

rdegges committed:
Adding changelog docs.

rdegges committed:
Upgrading tag manager / fb / optmizely scripts.

rdegges committed:
Version bump.

Commit:
Updated assertJsonResponse in test_views.

 - assertJsonResponse no longer assumes that all 'GET' requests will return
   a json representation of form field settings (now directly compares
   response data and expected data)
 - added another helper assert method that double checks expected form fields
   response to form field settings in the config file

Commit:
Minor updates to parameters when calling make_stormpath_response.

 - all data parameters now called as kwargs

rdegges committed:
Using real emails everywhere.

Commit:
Updated VerifyEmail view.

 - updated tests

Commit:
Added uuid to test social directory.

Commit:
Python 3 support.

 - added assertDictList to StormpathTestCase, for Python 3 sorting
   (using sorted() instead of sort())
 - added error message parsing to StormpathView
   (parses messages based on the presence of user_message)
 - stormpath_html_response_test now Python 3 compatible
 - updated account fetching in TestVerify setUp()
 - added decode() to json data
 - updated testing exceptions in test_models
 - added additional tests

Commit:
Updated dependencies (fixed deprecation warnings).
(Resolved issue #80)

  - removed `.ext` in imports, deprecated
  - replaced flask_wtf.Form with FlaskForm, deprecated
    (https://github.com/stormpath/stormpath-flask/issues/80)

Commit:
Updated dependencies in setup.py.

Commit:
Merge branch 'develop' of https://github.com/stormpath/stormpath-flask into temp_fork_cetko_backup

Conflicts:
	flask_stormpath/__init__.py
	flask_stormpath/context_processors.py
	flask_stormpath/forms.py
	flask_stormpath/views.py
	requirements.txt
	setup.py
	tests/test_context_processors.py
	tests/test_decorators.py
	tests/test_models.py
	tests/test_signals.py
	tests/test_views.py

Commit:
Fixed August week 3 issues.

  - accepted_types presence is checked via if not, not len() == 0
  - updated social view call validation
  - minor formatting changes

Commit:
Partial default-config update.
(We don't wanna lose our onw changes to the file, so we
updated manually.)

Commit:
Disabled CSRF protection on JSON requests.

Commit:
Renamed Facebook environment variables.

Commit:
Updated CSRF protection for Flask-WTF 0.14.

Commit:
Added capabilities for testing disabled views.

  - yaml config file now stored in application configuration, settable by user
  - added temporary tests/config directory for storing different yaml testing
    configuration files
  - added reinit method (needed for switching between different configs)
  - added tester helper method create_config_path
  - added tester helper method assertDisabledView for testing disabled views
    across test_views.py
  - updated dependencies (ruamel.yaml)

Commit:
Updated social error message in tests.

Commit:
Fixed wrong Facebook env variable name.

Commit:
Minor optimization.

Commit:
Added KeyboardInterrupt teardown.
